backend/de.metas.contracts/src/main/java/de/metas/contracts/modular/ModCntrInvoiceType.java

```diff
@@ -22,23 +22,25 @@
 
 package de.metas.contracts.modular;
 
-import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
 import lombok.Getter;
 import lombok.NonNull;
 import lombok.RequiredArgsConstructor;
 import org.compiere.model.X_C_DocType;
 
 import static de.metas.contracts.modular.ComputingMethodType.DEFINITIVE_INVOICE_SPECIFIC_METHODS;
 import static de.metas.contracts.modular.ComputingMethodType.FINAL_INVOICE_SPECIFIC_METHODS;
+import static de.metas.contracts.modular.ComputingMethodType.SALES_FINAL_INVOICE_SPECIFIC_METHODS;
 
 @Getter
 @RequiredArgsConstructor
 public enum ModCntrInvoiceType
 {
 	Final(X_C_DocType.DOCSUBTYPE_FinalInvoice, X_C_DocType.DOCSUBTYPE_FinalCreditMemo, FINAL_INVOICE_SPECIFIC_METHODS),
-	Definitive(X_C_DocType.DOCSUBTYPE_DefinitiveInvoice, X_C_DocType.DOCSUBTYPE_DefinitiveCreditMemo, DEFINITIVE_INVOICE_SPECIFIC_METHODS);
+	Definitive(X_C_DocType.DOCSUBTYPE_DefinitiveInvoice, X_C_DocType.DOCSUBTYPE_DefinitiveCreditMemo, DEFINITIVE_INVOICE_SPECIFIC_METHODS),
+	SalesFinal(X_C_DocType.DOCSUBTYPE_FinalInvoice, X_C_DocType.DOCSUBTYPE_FinalCreditMemo, SALES_FINAL_INVOICE_SPECIFIC_METHODS);
 
 	@NonNull final String positiveAmtDocSubType;
 	@NonNull final String negativAmtDocSubType;
-	@NonNull final ImmutableList<ComputingMethodType> computingMethodTypes;
+	@NonNull final ImmutableSet<ComputingMethodType> computingMethodTypes;
```

Comment:
positiveAmtDocSubType, negativAmtDocSubType and computingMethodTypes are never set.

Reply:
`@RequiredArgsConstructor` plus

```java
public enum ModCntrInvoiceType
{
	PurchaseFinal(X_C_DocType.DOCSUBTYPE_FinalInvoice, X_C_DocType.DOCSUBTYPE_FinalCreditMemo, FINAL_INVOICE_SPECIFIC_METHODS),
	Definitive(X_C_DocType.DOCSUBTYPE_DefinitiveInvoice, X_C_DocType.DOCSUBTYPE_DefinitiveCreditMemo, DEFINITIVE_INVOICE_SPECIFIC_METHODS),
	SalesFinal(X_C_DocType.DOCSUBTYPE_FinalInvoice, X_C_DocType.DOCSUBTYPE_FinalCreditMemo, SALES_FINAL_INVOICE_SPECIFIC_METHODS);
```

does the job